tremor-script/src/ast/deploy/raw.rs

```diff
@@ -0,0 +1,785 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// We want to keep the names here
+#![allow(clippy::module_name_repetitions)]
+
+use super::BaseExpr;
+use super::ConnectorDecl;
+use super::CreateStmt;
+use super::FlowDecl;
+use super::PipelineDecl;
+use super::Value;
+use crate::ast::error_generic;
+use crate::ast::query::raw::QueryRaw;
+use crate::ast::raw::ExprRaw;
+use crate::ast::raw::IdentRaw;
+use crate::ast::raw::ModuleRaw;
+use crate::ast::raw::StringLitRaw;
+use crate::ast::raw::WithExprsRaw;
+use crate::ast::AggrRegistry;
+use crate::ast::Deploy;
+use crate::ast::DeployStmt;
+use crate::ast::Helper;
+use crate::ast::NodeMetas;
+use crate::ast::Registry;
+use crate::ast::Upable;
+use crate::errors::ErrorKind;
+use crate::errors::Result;
+use crate::impl_expr;
+use crate::interpreter::Cont;
+use crate::pos::Location;
+use crate::AggrType;
+use crate::EventContext;
+use beef::Cow;
+use halfbrown::HashMap;
+use tremor_common::time::nanotime;
+use tremor_common::url::TremorUrl;
+use tremor_value::literal;
+use value_trait::Mutable;
+
+// For compile time interpretation support
+use crate::interpreter::Env;
+use crate::interpreter::ExecOpts;
+use crate::interpreter::LocalStack;
+
+pub type DeployWithExprsRaw<'script> = Vec<(IdentRaw<'script>, ExprRaw<'script>)>;
+
+/// Convenience to evaluate an expression at compile time so that
+/// evaluated results can be used during compilation
+#[macro_export]
+macro_rules! stateless_run_script {
```

Comment:
Could we turn this into a function? It looks like it doesn't require nay macroisms

Reply:
This actually started as a function / variant of `run` - but it went quickly downhill due to lifetime inconsistencies.
Also - its worth investigation if we can use `reduce` instead as well. Its worth another run to get these as functions/methods though and to remove the macros. Let's look into that tomorrow ( after coffee! ).